packages/cli/lib/charm.ts

```diff
@@ -477,20 +484,38 @@ export async function callCharmHandler(
   const manager = await loadManager(config);
   const charms = new CharmsController(manager);
   const charm = await charms.get(config.charm);
-  
+
   // Get the cell and traverse to the handler using .key()
   const cell = charm.getCell();
   const handlerStream = cell.key(handlerName);
-  
+
   // The handlerStream should be the actual stream object
   if (!handlerStream || !isStream(handlerStream)) {
     throw new Error(`Handler "${handlerName}" not found or not a stream`);
   }
-  
+
   // Send the event to trigger the handler
   handlerStream.send(args);
-  
+
   // Wait for processing to complete
   await manager.runtime.idle();
   await manager.synced();
 }
+
+/**
+ * Removes a charm from the space (moves it to trash).
+ */
+export async function removeCharm(
+  config: CharmConfig,
+): Promise<void> {
+  const manager = await loadManager(config);
+  const charms = new CharmsController(manager);
+  const removed = await charms.remove(config.charm);
+
+  if (!removed) {
```

Comment:
This isn't a validation error (e.g. improper input), can just throw a regular error here